UpcomingMovies/Stories/ListMovies/Controllers/Cell/ListMoviesTableViewCell.swift

```
@@ -0,0 +1,64 @@
+//
+//  ListMoviesTableViewCell.swift
+//  UpcomingMovies
+//
+//  Created by Luan Costa on 25/01/18.
+//  Copyright © 2018 Luan Costa. All rights reserved.
+//
+
+import UIKit
+
+class ListMoviesTableViewCell: UITableViewCell {
+    
+    @IBOutlet weak var movieImageView: UIImageView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var releaseDateLabel: UILabel!
+    
+    var viewModel: ViewModel? {
+        didSet {
+            didSetViewModel()
+        }
+    }
+    
+    override func draw(_ rect: CGRect) {
+        super.draw(rect)
+        
+        movieImageView.roundImageView()
+    }
+    
+    private func didSetViewModel() {
+        guard let title = viewModel?.title,
+            let releaseDate = viewModel?.releaseDate,
+            let posterImage = viewModel?.moviewImage else {
+            return
```

What happens when you don't have a view model?

---

If dont have any viewModel, the cell will behave as the way that was created on storyboard. So, the cell will present a empty image with no title and release date. In this case, I can treat the cell and let a friendly message to the user, showing that are no title, release date or image view for that movie or just hiding the cell on viewController. I maked this validation just to not let the application crash if there is no view model, in case that this cell can be used in a future for another controllers.